### PR TITLE
feat: Collapse component

### PR DESCRIPTION
**Pull request by zero0Halo**
feature(Collapse): Added some styles, Storybook knobs

feature(Collapse): Put in a theme system

Themes are individual files that use styled-components css module to render in full styles for specified themes

fix(collapse): Color for icons

Choosing different icons for storybook correctly applies color to them

chore(cleanup): Comments added, magic numbers removed

chore(cleanup): Removed svg color from themes

chore(test): Tests for Collapse are now 100% coverage

fix(ArrowBack): Corrected className

**Before submitting a pull request,** please make sure the following is done:

I have done **all** of the following:

- [x] Added a top level class to all my components `'.anchor-[COMPONENT NAME]'`.
- [x] Used [conventional commits](https://www.conventionalcommits.org) for all work.
- [x] Tested my solution on Mobile & Tablet.
- [x] Wrote [unit tests](https://jestjs.io/docs/en/getting-started) for states and all behavior (`npm test`) and passed coverage thresholds.
- [x] Updated snapshots for all permutations (`npm test -- -u`).
- [x] Accounted for hover, focus, blur, visited, & error states because they are not edge cases.
- [ ] Created TODOs for known edge cases.
- [x] Documented all of my changes (inline & doc site).
- [x] Made sure that all accessibility errors are resolved.
- [x] Added [stories](https://storybook.js.org/docs/basics/introduction/) with knobs for all possible configurations.
- [x] De-linted and ran [prettier](https://github.com/prettier/prettier) (`npm run pretty`) on my code.

---------
**Outline your feature or bug-fix below**

This adds the Collapse component to Anchor. This is an MVP which deploys some of the basic features wanted/needed but there is definitely room for more.

### CODE REVIEW

File: src/Collapse/Collapse.component.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    openText = DEFAULT_OPEN_TEXT,
+    closeText = DEFAULT_CLOSE_TEXT,
+    openIcon = false,
+    closeIcon = false,
+    theme = DEFAULT_THEME,
+    className,
+    children,
+}: CollapseProps) => {
+    const [open, toggleOpen] = React.useState(isOpen);
+    const IconOpen = openIcon ? openIcon : <ChevronDown />;
```
Comment (sharkparty) [inline src/Collapse/Collapse.component.tsx]: nominal note here. You can do this same assignment this way:
```const IconOpen = openIcon || <ChevronDown />;```

File: src/Collapse/Collapse.component.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    isOpen = false,
+    openText = DEFAULT_OPEN_TEXT,
+    closeText = DEFAULT_CLOSE_TEXT,
+    openIcon = false,
+    closeIcon = false,
+    theme = DEFAULT_THEME,
+    className,
+    children,
+}: CollapseProps) => {
+    const [open, toggleOpen] = React.useState(isOpen);
```
Comment (sharkparty) [inline src/Collapse/Collapse.component.tsx]: You can type this.

```
const [open, toggleOpen] = React.useState<boolean>(isOpen);
```

**Review by sharkparty — CHANGES_REQUESTED** [review]
Made some minor notes; nothing imperative.

File: src/Collapse/Collapse.component.tsx (hunk 2 above)
Comment (zero0Halo) [reply src/Collapse/Collapse.component.tsx]: Preferred that way as to enforce type?

Comment (sharkparty) [reply src/Collapse/Collapse.component.tsx]: Meh, more of a statement of intent that works as documentation. It obviously doesn't impact behavior, but it can help people from making silly mistakes.

**Review by sharkparty — APPROVED** [review]
Lgtm

**Review by brandonlilly — APPROVED** [review]
